src/nvim/ex_getln.c

```diff
@@ -3928,6 +3929,7 @@ nextwild (
   i = (int)(xp->xp_pattern - ccline.cmdbuff);
   assert(ccline.cmdpos >= i);
   xp->xp_pattern_len = (size_t)ccline.cmdpos - (size_t)i;
+  xp->xp_offset = 0;
```

Comment:
I would suggest this is OK for now, but add a TODO comment for it it could be simplified later (other expanders somehow do this without extra offset var).

Reply:
Do you know of other expansion contexts where this happens? `EXPAND_LUA` is special as far as I can tell in that the context needed for the completion (`xp->xp_pattern`) is longer than the actual part of `ccline.cmdlinebuff` that we are completing. This is primarily due to the completion of nested dicts. 

Other cases don't seem to have this problem, and built-in vimscript completion doesn't even try to address it: if you evaluate `:let b:tmp = {"a": {"b": {"hello": 5, "halla": 6}}}`, and try to complete `:echo b:tmp.a.b.h`, vim simply sets `xp->xp_pattern` to "h", and so looks up completion in the global namespace instead of in `b:tmp.a.b`. Similarly, if you try to complete `:echo g:`, vim sets `xp->xp_pattern` to "g:", and so all completion results will have a "g:" appended in front.

Reply:
it does handle it in some cases, like for nested directories. But we don't need to worry about the details now, I just mean we want to _consider_ to improve this situation later, so adding a TODO comment would be sensible.

Reply:
It turns out that it's actually a pretty simple change to not show the entire context; I've removed the xp_offset var in the newest commit.